src/apis.cpp

```diff
@@ -155,13 +179,16 @@ int gds_prepare_send(struct gds_qp *qp, struct ibv_exp_send_wr *p_ewr,
                         gds_dbg("ENOMEM error %d in ibv_exp_post_send\n", ret);
                 } else {
                         gds_err("error %d in ibv_exp_post_send\n", ret);
+                        //request not commited yet!
                 }
                 goto out;
         }
         
         ret = ibv_exp_peer_commit_qp(qp->qp, &request->commit);
         if (ret) {
                 gds_err("error %d in ibv_exp_peer_commit_qp\n", ret);
+                //request not commited in case of error
```

Comment:
this comment and the previous one seems to be a reminder... just document the function saying that the request will not be committed in case of errors.